src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1700,6 +1737,63 @@ impl ToQemuParams for ObjectIoThread {
     }
 }
 
+#[derive(Debug)]
+struct ObjectSevSnpGuest {
+    id: String,
+    cbitpos: u32,
+    reduced_phys_bits: u32,
+    kernel_hashes: bool,
+
+    is_snp: bool,
+    certs_path: String,
+}
+
+impl ObjectSevSnpGuest {
+    fn new(is_snp: bool, cbitpos: u32) -> Self {
+        ObjectSevSnpGuest {
+            id: (if is_snp { "snp" } else { "sev" }).to_owned(),
+            cbitpos,
+            reduced_phys_bits: 1,
```

Comment:
If `reduced_phys_bits` is hardcode, what's about removing this field `reduced_phys_bits` and just adding it `params.push(format!("reduced-phys-bits=1"));`. Or is there anything todo with it ?

Reply:
Frankly, I'm not sure.  runtime-go always puts 1 but since QEMU takes this as a parameter I assume it can take other values as well.  So I follow a general convention in the file where we represent every QEMU parameter with a corresponding structure member, and if the param happens to always take the same value, it's supplied in the structure's constructor (see e.g. `DeviceIntelIommu`, `DevicePciBridge` and others).